UPDATE SQL clause preparedstatement parsing does not always work correctly

## Driver version 
7.2 jre11

## SQL Server `version`
Any

## Client Operating System
Any Windows

## JAVA/JVM version
OpenJDK 11

## Problem description
For some reason only fist prepared statement query is parsed correctly
`UPDATE TxCounter SET addedCnt=addedCnt+1 OUTPUT INSERTED.addedCnt WHERE code=:S1 AND (cnt=-1 OR cnt=-2 OR addedCnt<cnt)
`
and
`UPDATE TxCounter SET addedCnt=addedCnt+1 OUTPUT INSERTED.addedCnt WHERE (cnt=-1 OR cnt=-2 OR addedCnt<cnt) AND code=:S1`
1. Expected behaviour: Both queries are parsed and works equal
2. Actual behaviour:  First prepared statement  parse does not find parameters at all
3. Error message/stack trace: 
While trying to set parameter value following error has been thrown
```
Caused by: com.microsoft.sqlserver.jdbc.SQLServerException: The index 1 is out of range.
	at com.microsoft.sqlserver//com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDriverError(SQLServerException.java:228)
	at com.microsoft.sqlserver//com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.setterGetParam(SQLServerPreparedStatement.java:1124)
	at com.microsoft.sqlserver//com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.setValue(SQLServerPreparedStatement.java:1138)
	at com.microsoft.sqlserver//com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement.setString(SQLServerPreparedStatement.java:1683)
```

PS: Both one works nicely in 7.1.4

## Reproduction code
```
PreparedStatement st1 = conn.prepareStatement(
                  "UPDATE TxCounter SET addedCnt=addedCnt+1 OUTPUT INSERTED.addedCnt WHERE code=? AND (cnt=-1 OR cnt=-2 OR addedCnt<cnt)");
PreparedStatement st2 = conn.prepareStatement(
                  "UPDATE TxCounter SET addedCnt=addedCnt+1 OUTPUT INSERTED.addedCnt WHERE (cnt=-1 OR cnt=-2 OR addedCnt<cnt) AND code=?");
st1.setString(1, "x");
//This one fails
st2.setString(1, "x");
```

--- a/src/main/java/com/microsoft/sqlserver/jdbc/ParameterUtils.java
+++ b/src/main/java/com/microsoft/sqlserver/jdbc/ParameterUtils.java
@@ -98,6 +98,10 @@ final class ParameterUtils {
                         break;
                     }
                     // Fall through to test character
+                default:
+                    if (ch == chTmp)
+                        return offset - 1;
+                    break;
                 case '[':
                     chTmp = ']';
                 case '\'':
@@ -111,10 +115,6 @@ final class ParameterUtils {
                             ++offset;
                         }
                     }
-                    break;
-                default:
-                    if (ch == chTmp)
-                        return offset - 1;
                     break;
             }
         }

--- a/src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/RegressionTest.java
+++ b/src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/RegressionTest.java
@@ -4,6 +4,7 @@
  */
 package com.microsoft.sqlserver.jdbc.preparedStatement;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.fail;
 
 import java.lang.reflect.Field;
@@ -371,6 +372,18 @@ public class RegressionTest extends AbstractTest {
         }
     }
 
+    @Test
+    public void testQueryParamsWithHyphen() throws Exception {
+        try (Connection con = DriverManager.getConnection(connectionString); 
+            PreparedStatement st1 = con.prepareStatement("SELECT 1 WHERE -1=-1 AND 1=?")){
+            st1.setInt(1, 1);
+            try (ResultSet rs = st1.executeQuery()) {
+                while (rs.next())
+                    assertEquals(1, rs.getInt(1));
+            }
+        }
+    }
+    
     /**
      * Cleanup after test
      * 

--- a/src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/RegressionTest.java
+++ b/src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/RegressionTest.java
@@ -383,6 +383,44 @@ public class RegressionTest extends AbstractTest {
             }
         }
     }
+
+    @Test
+    public void testQueryParamsWithComment() throws Exception {
+        try (Connection con = DriverManager.getConnection(connectionString); 
+            PreparedStatement st1 = con.prepareStatement("/**COMMENT**/ SELECT 1 WHERE 1=?")){
+            st1.setInt(1, 1);
+            try (ResultSet rs = st1.executeQuery()) {
+                while (rs.next())
+                    assertEquals(1, rs.getInt(1));
+            }
+        }
+    }
+
+    @Test
+    public void testQueryParamsWithLineComment() throws Exception {
+        try (Connection con = DriverManager.getConnection(connectionString); 
+            PreparedStatement st1 = con.prepareStatement("--comment\nSELECT 1 WHERE 1=?")){
+            st1.setInt(1, 1);
+            try (ResultSet rs = st1.executeQuery()) {
+                while (rs.next())
+                    assertEquals(1, rs.getInt(1));
+            }
+        }
+    }
+
+    @Test
+    public void testQueryParamsWithBackSlash() throws Exception {
+        try (Connection con = DriverManager.getConnection(connectionString); 
+            PreparedStatement st1 = con.prepareStatement("SELECT 1, '/''' AS str WHERE 1=?")){
+            st1.setInt(1, 1);
+            try (ResultSet rs = st1.executeQuery()) {
+                while (rs.next()) {
+                    assertEquals(1, rs.getInt(1));
+                    assertEquals("/'", rs.getString("str"));
+                }
+            }
+        }
+    }
     
     /**
      * Cleanup after test

